Module cannot be built and install on Ubuntu 20.04

An error occurs while compiling the DKMS module. Ubuntu 20.04, kernel 5.4.

```
# make
  CC [M]  /root/amneziawg-linux-kernel-module/src/main.o
In file included from <command-line>:
/root/amneziawg-linux-kernel-module/src/compat/compat.h:1291:19: error: static declaration of 'timer_delete' follows non-static declaration
 1291 | static inline int timer_delete(struct timer_list *timer)
      |                   ^~~~~~~~~~~~
In file included from ./include/linux/workqueue.h:9,
                 from ./include/linux/srcu.h:21,
                 from ./include/linux/notifier.h:16,
                 from ./arch/x86/include/asm/uprobes.h:13,
                 from ./include/linux/uprobes.h:49,
                 from ./include/linux/mm_types.h:14,
                 from ./include/linux/mmzone.h:21,
                 from ./include/linux/gfp.h:6,
                 from ./include/linux/mm.h:10,
                 from ./include/linux/bvec.h:13,
                 from ./include/linux/skbuff.h:17,
                 from ./include/linux/ip.h:16,
                 from ./include/linux/if_tunnel.h:5,
                 from ./include/net/ip_tunnels.h:5,
                 from ./include/net/udp_tunnel.h:5,
                 from /root/amneziawg-linux-kernel-module/src/compat/udp_tunnel/udp_tunnel_partial_compat.h:154,
                 from /root/amneziawg-linux-kernel-module/src/compat/compat.h:68,
                 from <command-line>:
./include/linux/timer.h:185:12: note: previous declaration of 'timer_delete' was here
  185 | extern int timer_delete(struct timer_list *timer);
      |            ^~~~~~~~~~~~
/root/amneziawg-linux-kernel-module/src/main.c: In function 'wg_mod_init':
/root/amneziawg-linux-kernel-module/src/main.c:25:13: error: implicit declaration of function 'chacha20_mod_init' [-Werror=implicit-function-declaration]
   25 |  if ((ret = chacha20_mod_init()) || (ret = poly1305_mod_init()) ||
      |             ^~~~~~~~~~~~~~~~~
/root/amneziawg-linux-kernel-module/src/main.c:25:44: error: implicit declaration of function 'poly1305_mod_init' [-Werror=implicit-function-declaration]
   25 |  if ((ret = chacha20_mod_init()) || (ret = poly1305_mod_init()) ||
      |                                            ^~~~~~~~~~~~~~~~~
/root/amneziawg-linux-kernel-module/src/main.c:26:13: error: implicit declaration of function 'chacha20poly1305_mod_init'; did you mean 'chacha20poly1305_decrypt'? [-Werror=implicit-function-declaration]
   26 |      (ret = chacha20poly1305_mod_init()) || (ret = blake2s_mod_init()) ||
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~
      |             chacha20poly1305_decrypt
/root/amneziawg-linux-kernel-module/src/main.c:26:52: error: implicit declaration of function 'blake2s_mod_init'; did you mean 'blake2s_init'? [-Werror=implicit-function-declaration]
   26 |      (ret = chacha20poly1305_mod_init()) || (ret = blake2s_mod_init()) ||
      |                                                    ^~~~~~~~~~~~~~~~
      |                                                    blake2s_init
/root/amneziawg-linux-kernel-module/src/main.c:27:13: error: implicit declaration of function 'curve25519_mod_init' [-Werror=implicit-function-declaration]
   27 |      (ret = curve25519_mod_init()))
      |             ^~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:270: /root/amneziawg-linux-kernel-module/src/main.o] Error 1
make[1]: *** [Makefile:1783: /root/amneziawg-linux-kernel-module/src] Error 2
make: *** [Makefile:27: module] Error 2

```

--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -29,6 +29,8 @@
 #define ISUBUNTU1904
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
 #define ISUBUNTU1910
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+#define ISUBUNTU2004
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
 #define ISUBUNTU2204
 #endif
@@ -892,7 +894,7 @@ static inline void skb_mark_not_on_list(struct sk_buff *skb)
 #endif
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 200) || (LINUX_VERSION_CODE < KERNEL_VERSION(4, 20, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 249)) || (LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 285)) || (LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 320))) && LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 200) || (LINUX_VERSION_CODE < KERNEL_VERSION(4, 20, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 249)) || (LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 285)) || (LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 320))) && LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0) && !defined(ISUBUNTU2004)
 #define COMPAT_INIT_CRYPTO
 #define blake2s_init zinc_blake2s_init
 #define blake2s_init_key zinc_blake2s_init_key
@@ -1286,7 +1288,7 @@ static inline void dev_sw_netstats_rx_add(struct net_device *dev, unsigned int l
 }
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 91) && !defined(ISUBUNTU2204) && !defined(ISRHEL9)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 91) && !defined(ISUBUNTU2004) && !defined(ISUBUNTU2204) && !defined(ISRHEL9)
 #include <linux/timer.h>
 static inline int timer_delete(struct timer_list *timer)
 {

